### PR TITLE
Fend 332 add new standards rules

### PR DESCRIPTION
**Pull request by daltonwignall**
#### Summary

I updated our ESLint config based on our new set of code standards to make many of them auto enforced. Additionally I added some new stylistic rules to assure that our code looks consistent across platforms. I will be upgrading the version to 4.0.0 so that it doesn't auto update for our projects and cause any pipelines to fail.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,12 @@ This will lead to poor performance. Use `Promise.all()` instead.
 #### [no-compare-neg-zero](http://eslint.org/docs/rules/no-compare-neg-zero)\*
 Comparing to `-0` will pass for 0 and -0 which is confusing.
 
+### [no-cond-assign](https://eslint.org/docs/rules/no-cond-assign)
```
Comment (alyssabrunswick) [inline README.md]: should be 4 hashes?

**Review by alyssabrunswick — COMMENTED** [review]
HR on line 24 above Best Practices, 310 above Strict Mode, 315 above Variables, Stylistic Issues, ECMAScript 6, React, JSX

Rather than those headings being links, link under them, so that the heading stands out more, then make these internal links that are listed at the top of the readme

File: README.md (hunk 1 above)
Comment (daltonwignall) [reply README.md]: Good catch, see: 565ab01937cc622e1eda88250d4a61bdfbcec339

**Comment by daltonwignall** [comment]
See: 565ab01937cc622e1eda88250d4a61bdfbcec339 for the hr comment

**Pull request merged**